src/test/java/com/cloudant/tests/SearchTests.java

```diff
@@ -202,6 +201,25 @@ public void drillDownTest() {
         assertEquals(rslt.getRanges().get("min_length").get("large"), new Long(0));
     }
 
+    @Test
+    public void multipleDrillDownTest() {
+        // do a faceted search for drilldown
+        Search srch = db.search("views101/animals");
+        SearchResult<Animal> rslt = srch.includeDocs(true)
+                .counts(new String[]{"class", "diet"})
+                .ranges("{ \"min_length\": {\"small\": \"[0 TO 1.0]\","
+                        + "\"medium\": \"[1.1 TO 3.0]\", \"large\": \"[3.1 TO 9999999]\"} }")
+                .drillDown("class", "bird")
+                .drillDown("diet", "omnivore")
+                .querySearchResult("class:bird", Animal.class);
+        assertNotNull(rslt);
+        assertNotNull(rslt.getRanges());
+        assertEquals(rslt.getRanges().entrySet().size(), 1);
```

Comment:
I think JUnit assertions are `assert*(expected, actual)` so I think these args (and for the following asserts) might need to be reversed to avoid confusing failure messages.

I think it is probably also worth having a comment on this test explaining how it works. I'm assuming that having two drilldowns limits the returned result to a single small animal?